pan3d/explorers/analytics.py

```diff
@@ -0,0 +1,576 @@
+    vtkRenderer,
+    vtkRenderWindowInteractor,
+    vtkRenderWindow,
+    vtkActor,
+    vtkPolyDataMapper,
+)
+from vtkmodules.vtkFiltersGeometry import vtkDataSetSurfaceFilter
+
+# VTK factory initialization
+from vtkmodules.vtkInteractionStyle import vtkInteractorStyleSwitch  # noqa
+import vtkmodules.vtkRenderingOpenGL2  # noqa
+
+import json
+import traceback
+from pathlib import Path
+
+from trame.decorators import TrameApp, change
+from trame.app import get_server, asynchronous
+
+from trame.ui.vuetify3 import VAppLayout
+from trame.widgets import vuetify3 as v3, html
+
+from pan3d.xarray.algorithm import vtkXArrayRectilinearSource
+
+from pan3d.utils.constants import has_gpu
+from pan3d.utils.convert import update_camera, to_image, to_float
+from pan3d.utils.presets import set_preset
+
+from pan3d.ui.vtk_view import Pan3DView, Pan3DScalarBar
+from pan3d.ui.preview import SummaryToolbar, ControlPanel
+from pan3d.ui.analytics import Plotting
+
+"""
+@TrameApp()
+class Pan3dAnalyticsView(Pan3DView):
+    def __init__(self, render_window, **kwargs):
+        super().__init__(render_window=render_window, **kwargs)
+        with self.toolbar:
+            v3.VDivider(classes="my-1")
+            with v3.VTooltip(text="Look toward at an angle"):
+                with html.Template(v_slot_activator="{ props }"):
+                    v3.VBtn(
+                        v_bind="props",
+                        flat=True,
+                        density="compact",
+                        icon="mdi-axis-arrow",
+                        click=(self.reset_camera_to_axis, "[[1,1,1]]"),
+                    )
+"""
+
+
+@TrameApp()
+class AnalyticsExplorer:
+    """Create a Trame GUI for a Pan3D XArray Viewer"""
+
+    def __init__(self, xarray=None, server=None, local_rendering=None):
+        """Create an instance of the XArrayViewer class.
+
+        Parameters:
+            server (str/server): Trame server name or instance.
+            local_rendering (str): If provided (wasm, vtkjs) local rendering will be used
+
+        CLI options:
+            - `--import-state`: Pass a string with this argument to specify a startup configuration.
+                              This value must be a local path to a JSON file which adheres to the
+                              schema specified in the [Configuration Files documentation](../api/configuration.md).
+                              A dataset specified in this configuration will override any value passed to `--xarray-*`
+            - `--xarray-file`: Provide path to xarray file
+            - `--xarray-url`: Provide URL to xarray dataset
+            - `--wasm`: Use WASM for local rendering
+            - `--vtkjs`: Use vtk.js for local rendering
+        """
+        self.xarray = xarray
+        self.server = get_server(server, client_type="vue3")
+        if self.server.hot_reload:
+            self.ctrl.on_server_reload.add(self._build_ui)
+
+        # cli
+        self.server.cli.add_argument(
```

Comment:
we've never done that in the previous apps, but it might be good for the CLI to create a group for those add-on args so they don't get bundled with the rest of the trame/wslink ones. The idea, will be that when we run with `--help` we can clearly identify the ones that are for that explorer/app.